### ISSUE TITLE
An error occurs when the route optional matches query

### ISSUE THREAD

**Issue by bosens-China** [honojs/hono#2884]
### What version of Hono are you using?

4.3.11

### What runtime/platform is your app running on?

node

### What steps can reproduce the bug?

``` js
const app = new Hono()
  .get(
    "/filtration/:type?",
    zValidator(
      "query",
      z.object({
        pageSize: z.string().default("10"),
        page: z.string().default(""),
      })
    ),
    async (c) => {
      const { pageSize, page } = c.req.valid("query");


      return c.json({});
    }
  );

export default app;

```

Work with clients

``` js
  const list = await client.api.issues.filtration[":type?"]
    .$get({ param: { type }, query: { page } })
    .then((res) => res.json());

```

result
query parsing error because :type? Suffix? The cause of the removal? All is well

**Comment by mgrithm** [comment]
hey @bosens-China 

could you give me more information about your problem?

**Comment by EdamAme-x** [comment]
please tell me details of error

**Comment by bosens-China** [comment]
[bug url](https://codesandbox.io/p/devbox/bug-tw6ykf?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clx69s4ha0007356ljm210b3y%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clx69s4ha0002356ll7c0fs3c%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clx69s4ha0004356lil3a32vu%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clx69s4ha0006356lr8cwzbj7%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clx69s4ha0002356ll7c0fs3c%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clx69s4h90001356laaoeozhx%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%257D%255D%252C%2522id%2522%253A%2522clx69s4ha0002356ll7c0fs3c%2522%252C%2522activeTabId%2522%253A%2522clx69s4h90001356laaoeozhx%2522%257D%252C%2522clx69s4ha0006356lr8cwzbj7%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clx69s4ha0005356lt9fp2noh%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522start%2522%252C%2522port%2522%253A3000%252C%2522path%2522%253A%2522%2522%257D%255D%252C%2522id%2522%253A%2522clx69s4ha0006356lr8cwzbj7%2522%252C%2522activeTabId%2522%253A%2522clx69s4ha0005356lt9fp2noh%2522%257D%252C%2522clx69s4ha0004356lil3a32vu%2522%253A%257B%2522id%2522%253A%2522clx69s4ha0004356lil3a32vu%2522%252C%2522activeTabId%2522%253A%2522clx69s4ha0003356l6fvpla2g%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clx69s4ha0003356l6fvpla2g%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522start%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522yarn%2520add%2520hono%25404.4.2%2522%252C%2522id%2522%253A%2522clx6a5oq600gn356ld65eqmbf%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522yarn%2520add%2520zod%25403.23.8%2522%252C%2522id%2522%253A%2522clx6a5pdq00ic356liky3pe83%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522yarn%2520add%2520%2540hono%252Fzod-validator%25400.2.2%2522%252C%2522id%2522%253A%2522clx6a5pj700k1356l02bill0c%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522yarn%2520add%2520%2540hono%252Fnode-server%25401.11.2%2522%252C%2522id%2522%253A%2522clx6a5q8500lq356lz072z0ha%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Comment by yusukebe** [comment]
Hi @bosens-China 

This issue has been fixed with the latest version, `4.4.4. Can you try it?
